src/vos/vos_aggregate.c

```diff
@@ -239,6 +240,14 @@ vos_agg_obj(daos_handle_t ih, vos_iter_entry_t *entry,
 	}
 
 	if (agg_param->ap_discard) {
+		if (agg_param->ap_sub_tree_empty) {
+			struct vos_obj_iter	*oiter = vos_hdl2oiter(ih);
+			struct vos_object	*obj = oiter->it_obj;
+
+			D_ASSERT(daos_unit_obj_id_equal(obj->obj_id, oid));
```

Comment:
This isn't correct, the 'ap_oid' in agg_param is the last processed oid, not necessarily equal to current object oid.